### PR TITLE
More-typescript-2

### PR DESCRIPTION
**Pull request by wolverineks**

### CODE REVIEW

**Comment by wolverineks** [comment]
strictly a refactor,
no ui changes

File: src/Settings/ChangePassword/ChangePasswordScreen.tsx [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
+import {Button, Checkmark, Spacer, TextInput} from '../../components'
 import {useSelectedWallet} from '../../SelectedWallet'
 import {WalletInterface} from '../../types'
 
 export const ChangePasswordScreen = () => {
   const strings = useStrings()
   const navigation = useNavigation()
 
-  const currentPasswordRef = React.useRef<{focus: () => void} | null>(null)
+  const currentPasswordRef = React.useRef<RNTextInput>(null)
```
Comment (wolverineks) [inline src/Settings/ChangePassword/ChangePasswordScreen.tsx:19]: react native paper issue (will not fix)
https://github.com/callstack/react-native-paper/issues/1322#issuecomment-699166622

File: src/Settings/RemoveWallet/RemoveWalletScreen.tsx [hunk 2]
```diff
@@ -74,7 +72,7 @@ export const RemoveWalletScreen = () => {
 
         <Spacer height={16} />
 
-        <Button onPress={removeWallet} title={strings.remove} style={styles.removeButton} disabled={disabled} />
+        <Button onPress={() => removeWallet()} title={strings.remove} style={styles.removeButton} disabled={disabled} />
```
Comment (wolverineks) [inline src/Settings/RemoveWallet/RemoveWalletScreen.tsx:75]: found a bug, YAY!

File: src/components/ProgressCircle.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+type ExternalProps = {
+  percentage: number
+}
+
+export const ProgressCircle = ({percentage}: ExternalProps) => (
+  <View style={styles.wrapper}>
+    <Text style={styles.textWrapper}>{percentage}%</Text>
+    {/* eslint-disable-next-line @typescript-eslint/no-explicit-any */}
+    <Svg viewBox="0 0 36 36" height={undefined as any} width={undefined as any} style={styles.chart}>
```
Comment (wolverineks) [inline src/components/ProgressCircle.tsx:16]: we use a really old version react-native-svg with bad types

File: src/components/TextInput/TextInput.tsx [hunk 4]
```diff
@@ -47,8 +56,8 @@ const useDebounced = (callback, value, delay = 1000) => {
   }, [callback, delay, value])
 }
 
-const TextInputWithRef = (
-  {
+export const TextInput = React.forwardRef((props: Props, ref: ForwardedRef<RNTextInput>) => {
```
Comment (wolverineks) [inline src/components/TextInput/TextInput.tsx]: eslint bug
https://github.com/yannickcr/eslint-plugin-react/issues/3140#issuecomment-1024897971

File: src/components/ProgressCircle.tsx (hunk 3 above)
Comment (stackchain) [reply src/components/ProgressCircle.tsx:16]: I tried to bump it a little time ago but I found some compilation issues, didn't make a big effort though, perhaps we "must" bump it

**Review by stackchain — APPROVED** [review]

**Pull request merged**